src/yoroi-wallets/cardano/utxoStorage.ts

```diff
@@ -0,0 +1,136 @@
+import BigNumber from 'bignumber.js'
+
+import storageLegacy from '../../legacy/storage'
+
+// yoroi-lib copy
+export type Asset = {
+  assetId: string
+  policyId: string
+  name: string
+  amount: string
+}
+
+// yoroi-lib copy
+export type Utxo = {
+  utxoId: string
+  txHash: string
+  txIndex: number
+  receiver: string
+  amount: BigNumber
+  assets: Asset[]
+  blockNum: number
+}
+
+// yoroi-lib copy
+export type UtxoAtSafePoint = {
+  lastSafeBlockHash: string
+  utxos: Utxo[]
+}
+
+// yoroi-lib copy
+export type UtxoDiffToBestBlock = {
+  lastBestBlockHash: string
+  spentUtxoIds: string[]
+  newUtxos: Utxo[]
+}
+
+// yoroi-lib copy
+export interface UtxoStorage {
+  getUtxoAtSafePoint(): Promise<UtxoAtSafePoint | undefined>
+  getUtxoDiffToBestBlock(): Promise<UtxoDiffToBestBlock[]>
+  replaceUtxoAtSafePoint(utxos: Utxo[], safeBlockHash: string): Promise<void>
+  clearUtxoState(): Promise<void>
+  appendUtxoDiffToBestBlock(diff: UtxoDiffToBestBlock): Promise<void>
+  removeDiffWithBestBlock(blockHash: string): Promise<void>
+}
+
+export type UtxoStorageItem = {
+  utxoAtSafePoint: UtxoAtSafePoint
+  utxoDiffToBestBlock: UtxoDiffToBestBlock[]
+}
+
+export class UtxoStorage implements UtxoStorage {
```

Comment:
Influenced by https://github.com/Emurgo/yoroi-lib/blob/a86e3c7b96868e610acafa978c64da6498295d63/samples/shared/json-storage.ts

Reply:
feel free to make this a factory function